src/Microsoft.Data.SqlClient/tests/FunctionalTests/SqlConnectionTest.cs

```diff
@@ -547,6 +547,95 @@ public void GetSchema_Connection_Closed()
             }
         }
 
+        [Theory]
+        [InlineData(@"AttachDbFileName=C:\test\attach.mdf", @"AttachDbFileName=C:\test\attach.mdf")]
+        [InlineData(@"AttachDbFileName=C:\test\attach.mdf;", @"AttachDbFileName=C:\test\attach.mdf;")]
+        public void ConnectionString_AttachDbFileName_Plain(string value, string expected)
+        {
+            SqlConnection cn = new SqlConnection();
+            cn.ConnectionString = value;
+            Assert.Equal(expected, cn.ConnectionString);
+        }
+
+        [Theory]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [InlineData(@"Data Source=.;AttachDbFileName=|DataDirectory|\attach.mdf",
+                    @"Data Source=.;AttachDbFileName=|DataDirectory|\attach.mdf",
+                    @"C:\test\")]
+        [InlineData(@"Data Source=.;AttachDbFileName=|DataDirectory|\attach.mdf",
+                    @"Data Source=.;AttachDbFileName=|DataDirectory|\attach.mdf",
+                    @"C:\test")]
+        [InlineData(@"Data Source=.;AttachDbFileName=|DataDirectory|attach.mdf",
+                    @"Data Source=.;AttachDbFileName=|DataDirectory|attach.mdf",
+                    @"C:\test")]
+        [InlineData(@"Data Source=.;AttachDbFileName=|DataDirectory|attach.mdf",
+                    @"Data Source=.;AttachDbFileName=|DataDirectory|attach.mdf",
+                    @"C:\test\")]
+        [InlineData(@"Data Source=.;AttachDbFileName=C:\test\attach.mdf;AttachDbFileName=|DataDirectory|attach.mdf",
+                    @"Data Source=.;AttachDbFileName=C:\test\attach.mdf;AttachDbFileName=|DataDirectory|attach.mdf",
+                    null)]
+        public void ConnectionString_AttachDbFileName_DataDirectory(string value, string expected, string dataDirectory)
+        {
+            AppDomain.CurrentDomain.SetData("DataDirectory", dataDirectory);
+
+            SqlConnection cn = new SqlConnection();
+            cn.ConnectionString = value;
```

Comment:
Can we also test `SqlConnectionStringBuilder` ?

Reply:
I tried